fleet/agent_group_service.go

```diff
@@ -110,12 +153,33 @@ func (svc fleetService) CreateAgentGroup(ctx context.Context, token string, s Ag
 		return AgentGroup{}, errors.Wrap(ErrCreateAgentGroup, err)
 	}
 
-	s.ID = id
-	err = svc.addAgentsToAgentGroupChannel(token, s)
+	ag, err := svc.agentGroupRepository.RetrieveByID(ctx, id, mfOwnerID)
+	if err != nil {
+		return AgentGroup{}, errors.Wrap(ErrCreateAgentGroup, err)
+	}
+
+	err = svc.addAgentsToAgentGroupChannel(token, ag)
 	if err != nil {
 		// TODO should we roll back?
 		svc.logger.Error("error adding agents to group channel", zap.Error(errors.Wrap(ErrMaintainAgentGroupChannels, err)))
 	}
 
-	return s, err
+	return ag, err
+}
+
+func (svc fleetService) RemoveAgentGroup(ctx context.Context, token, groupId string) error {
+	ownerID, err := svc.identify(token)
+	if err != nil {
+		return err
+	}
+	err = svc.agentGroupRepository.Delete(ctx, groupId, ownerID)
+	if err != nil {
+		return err
+	}
+
+	err = svc.removeAgentGroupSubscriptions(groupId, ownerID)
```

Comment:
doesn't this have to happen before the repository delete? removeAgentGroupSubscriptions calls RetrieveAllByAgentGroupID on the repo which will be empty because the group id has been deleted, right?